### DISCUSSION TITLE
Problems with a simple importing of an external Vue (Javascript) Component

### DISCUSSION THREAD

**Discussion by Pawel2505** [nuxt/nuxt#17384] [General]
I am struggling with adding a 3rd party component [vue-cookie-comply](https://github.com/yaiks/vue-cookie-comply) to a fresh Nuxt3 project.

I import it into my component as with: 
`import VueCookieComply from 'vue-cookie-comply'`
and my vscode warns me with this message:
> Could not find a declaration file for module 'vue-cookie-comply'. '.../node_modules/vue-cookie-comply/dist/vue-cookie-comply.umd.js' implicitly has an 'any' type. Try `npm i --save-dev @types/vue-cookie-comply` if it exists or add a new declaration (.d.ts) file containing `declare module 'vue-cookie-comply';`

Of course the command `npm i --save-dev @types/vue-cookie-comply` didn't work. 

I also tried to add the code below to my tsconfig.json - no result:
```
"compilerOptions": {
        "noImplicitAny": false,
        "allowJs": true,
    }
```
I think it is a very common scenario and I suppose the problem is that vue-cookie-comply component is written in JavaScript. However still don't know how to solve it :(

**Comment by Diizzayy** [comment]
@Pawel2505 The component just needs to be properly installed via a plugin. Have a look at the documentation for [using Vue plugins with Nuxt 3](https://v3.nuxtjs.org/guide/directory-structure/plugins#vue-plugins).


Remove any other changes you may have made to your `tsconfig.json` or elsewhere and create a new file as seen below.

`nuxt-app/plugins/cookie-comply.client.ts`
```ts

import VueCookieComply from 'vue-cookie-comply'
import 'vue-cookie-comply/dist/style.css'

export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.vueApp.use(VueCookieComply)
})
```

- usage

```vue
<template>
  <div>
    <ClientOnly>
      <vue-cookie-comply :preferences="preferences" />
    </ClientOnly>
  </div>
</template>
```

**Reply by Pawel2505** [reply]
Thank you very much for your rapid and correct answer!

The funny thing is that although I didn't mention I read the docs and I tried with plugins before - but it didn't work. I got error about weak references etc. Now again it did not work initially, however I restarted npm and now it is working! You saved me hours! Thank you!

**Reply by Diizzayy** [reply]
@Pawel2505 no problem at all, I'm glad to hear it's working for you now.
